app/controllers/api/base_controller/renderer.rb

```diff
@@ -178,7 +178,12 @@ def virtual_attribute_accessor(type, attr)
       private
 
       def resource_search(id, type, klass)
-        target = respond_to?("find_#{type}") ? public_send("find_#{type}", id) : klass.find(id)
+        target = if respond_to?("find_#{type}")
+                   public_send("find_#{type}", id)
+                 else
+                   raise NotFound, "Cound not find #{klass} id=#{id}" unless id.kind_of?(Integer) || id =~ /\A\d+\z/
```

Comment:
Unfortunately, we need both. When the resource_search is called from generic/manager we get Integer, however when custom actions call this we may get String.